x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/processors/config_driven/configs/user_agent.tsx

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiCode, EuiLink } from '@elastic/eui';
+import { UserAgentProcessorConfig, UserAgentProcessorDefinition } from '@kbn/streams-schema';
+import { isEmpty } from 'lodash';
+import { ConfigDrivenProcessorConfiguration } from '../types';
+
+export type UserAgentProcessorFormState = UserAgentProcessorConfig & { type: 'user_agent' };
+
+export const userAgentProcessorConfig: ConfigDrivenProcessorConfiguration<
+  UserAgentProcessorFormState,
+  UserAgentProcessorDefinition
+> = {
+  value: 'user_agent' as const,
+  inputDisplay: i18n.translate(
+    'xpack.streams.streamDetailView.managementTab.enrichment.processor.UserAgentInputDisplay',
+    {
+      defaultMessage: 'User agent',
+    }
+  ),
+  getDocUrl: (esDocUrl: string) => {
+    return (
+      <FormattedMessage
+        id="xpack.streams.streamDetailView.managementTab.enrichment.processor.userAgentHelpText"
+        defaultMessage="{userAgentLink} extracts details from the user agent string a browser sends with its web requests. This processor adds this information by default under the {userAgentField} field."
+        values={{
+          userAgentLink: (
+            <EuiLink
+              data-test-subj="streamsAppAvailableProcessorsUserAgentLink"
+              external
+              target="_blank"
+              href={esDocUrl + 'user-agent-processor.html'}
+            >
+              {i18n.translate('xpack.streams.availableProcessors.userAgentLinkLabel', {
+                defaultMessage: 'The user_agent processor',
+              })}
+            </EuiLink>
+          ),
+          userAgentField: <EuiCode>user_agent</EuiCode>,
+        }}
+      />
+    );
+  },
+  defaultFormState: {
+    type: 'user_agent' as const,
+    field: '',
+    target_field: '',
+    regex_file: '',
+    properties: [],
+    ignore_missing: false,
+  },
+  convertFormStateToConfig: (formState) => {
+    return {
+      user_agent: {
+        field: formState.field,
+        target_field: isEmpty(formState.target_field) ? undefined : formState.target_field,
```

Comment:
not sure whether this is overdoing it, but it seems like the only thing the convertXtoY functions are doing is an `isEmpty` check one way and filling in the default value the other way - we could move this to the generic part instead of implementing for each processor.

No strong opinion on it though, it just stood out a bit to me.

Reply:
Yeah, it doesn't really do anything beyond that with the field types available at the moment. I can look at just making that generic.

> I think it makes sense to keep this auto-generation logic very simple and prefer breaking out specific processors with a standalone component rather than making the generation more flexible.

Cool, that's fair 👍

Reply:
A couple of reasons why there was a specific mapper for each processor on the first processors:
- For some cases, it's not just an `isEmpty` check, but there is a data transformation, like for the Grok patterns.
- The form state has not the same shape as the config, where the values are nested as 
```
{
  [type]: {
    // ...
  }
}
```
I'd rather keep a 1:1 mapping for easier maintainability vs a generic mapper that does half the job and is less flexible for future changes.

Reply:
The non-config driven ones have [a function](https://github.com/elastic/kibana/blob/da8fb954d3e37c9abccb0767544afe2c71098116/x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/utils.ts#L144) and we definitely need to keep it that way there, I'm just suggesting this for the config driven ones.

The idea is that the config driven logic is kept simple on purpose and if you want to do Special Stuff, you need to "eject".